File: C#/BubbleSort/BubbleSort.cs
using System.Diagnostics;

namespace Algorithms
{
    public class BubbleSort
    {
        //sort given array in ascending order
        public int SortAscending(int[] array) 
        {
            bool isSorted = false;
            int counter = 0;

            //iterate over list until sorted
            while (!isSorted) 
            {
                counter++;
                isSorted = true;
                //check if each element is lower/higher than element to its right
                //swap if comparison met
                for (int i = 0; i < (array.Length - 1) - (counter - 1); i++)
                {
                    if (array[i] > array[i + 1])
                    {
                        int temp = array[i + 1];
                        array[i + 1] = array[i];
                        array[i] = temp;
                        isSorted = false;
                    }
                }
            }
            return counter;
        }

        //sort given array in descending order
        public int SortDescending(int[] array)
        {
            bool isSorted = false;
            int counter = 0;

            //iterate over list until sorted
            while (!isSorted)
            {
                counter++;
                isSorted = true;
                //check if each element is lower/higher than element to its right
                //swap if comparison met
                for (int i = 0; i < (array.Length - 1) - (counter - 1); i++)
                {
                    if (array[i] < array[i + 1])
                    {
                        int temp = array[i + 1];
                        array[i + 1] = array[i];
                        array[i] = temp;
                        isSorted = false;
                    }
                }
            }
            return counter;
        }
    }
}

File: C++/BubbleSort/BubbleSort.cpp
#include "BubbleSort.h"
#include <vector>
using namespace std;

//sorts given array using BubbleSort algorithm in ascending order
int BubbleSort::SortAscending(int arr[], int size)
{
	bool isSorted = false;
	int counter = 0;

	//repeat algorithm until array is sorted
	while (!isSorted) 
	{
		counter++;
		isSorted = true;
		//for each elements in array, compare element to its next element and adjust
		//position in array based on comparison (low/high)
		for (int i = 0; i < (size - 1) - (counter - 1); i++)
		{
			if (arr[i] > arr[i + 1]) 
			{
				swap(arr[i], arr[i + 1]);
				isSorted = false;
			}
		}
	}
	return counter;
}

//sorts given array using BubbleSort algorithm in descending order
int BubbleSort::SortDescending(int arr[], int size)
{
	bool isSorted = false;
	int counter = 0;

	//repeat algorithm until array is sorted
	while (!isSorted)
	{
		counter++;
		isSorted = true;
		//for each elements in array, compare element to its next element and adjust
		//position in array based on comparison (low/high)
		for (int i = 0; i < (size - 1) - (counter - 1); i++)
		{
			if (arr[i] < arr[i + 1])
			{
				swap(arr[i], arr[i + 1]);
				isSorted = false;
			}
		}
	}
	return counter;
}
